### DISCUSSION TITLE
ARIS Smart Designer (EPC)

### DISCUSSION THREAD

**Discussion by bpmbpm** [AlexeyBoiko/DgrmJS#26] [Ideas]
Есть ARIS Smart Designer (EPC):
https://www.ariscommunity.com/videos/how-model-smartdesign-process 
Хотелось бы его повторить на чем-то кроме ARIS Express, например, DgrmJS. 
Подготовил макет (прилагаю) Excel 2016 + Drawio в части Event + Function (только два объекта из EPC, чтобы идея была проще видна). Такое повторить бы в реальном времени, как в ARIS Smart Designer: автоматическое обновление, таблицу размещать непосредственно рядом со схемой процесса. Была бы интересна интеграция "встроенной" таблицы с Excel и Google Sheets. 
К макету: нужно указать пути (Drawio + Temp) на листе «Set» (нужно локально поставить Drawio, можно portable). Далее на листе List1 нажимать «в паре» кнопки: «Выгрузка в файл» и «Запуск файла». Также можно нажимать и другие кнопки, которые создают новые или удаляют Event + Function. Это функции: добавить объект (Event or Function) вниз (Добавление), вставить объект в указанное место (выбранную строку, Вставка), удалить конкретный объект (выбранную строку), очистить (удалить все). В ARIS Express добавляется объект только в конец, а потом перемещается вручную кнопками "вверх" \ "вниз". В макете показан более удобный вариант. 
Пользователю будут отображаться только столбцы с названиями Event & Function (остальные строки в Excel можно скрыть).
Макет на Excel 2016:
https://docs.google.com/spreadsheets/d/1xMdCPHQUOQ5Qx0HT4bAwgEU3MbTozAla/edit?usp=sharing&ouid=111910665780998526150&rtpof=true&sd=true
Половина кода - файловые преобразования: чтобы русские буквы были в Drawio, пришлось из стандартного Excel ASCII делать Unicode noBOM.

**Comment by AlexeyBoiko** [comment]
Добрый день,
простите за долгий ответ.

Вы хотите генерировать схему по описанию в Excel?

**Reply by bpmbpm** [reply]
Я тоже не очень быстрый ...
Да, я хочу генерировать схемы из описания в excel.

**Reply by AlexeyBoiko** [reply]
Добрый день,
Спасибо за идею. В ближайшее время такую возможность не планирую.
